components/datetime/src/time_zone.rs

```diff
@@ -251,61 +360,239 @@ impl TimeZoneFormat {
+    {
+        let locale = locale.into();
+        let units = SmallVec::<[TimeZoneFormatUnit; 6]>::new();
+
+        let data_payloads = TimeZoneDataPayloads {
+            zone_formats: zone_provider
+                .load_resource(&DataRequest {
+                    options: locale.clone().into(),
+                    metadata: Default::default(),
+                })?
+                .take_payload()?,
+            exemplar_cities: None,
+            mz_generic_long: None,
+            mz_generic_short: None,
+            mz_specific_long: None,
+            mz_specific_short: None,
+        };
 
         Ok(Self {
-            kind: TimeZoneFormatKind::Config(config),
-            zone_formats,
-            exemplar_cities,
-            mz_generic_long,
-            mz_generic_short,
-            mz_specific_long,
-            mz_specific_short,
+            kind: TimeZoneFormatKind::Config(TimeZoneFormatConfig::SpecificNonLocationLong),
+            data_payloads,
+            locale,
+            units,
         })
     }
 
+    pub fn load_generic_non_location_long<ZP>(
+        &mut self,
+        zone_provider: &ZP,
+    ) -> Result<&mut TimeZoneFormat, DateTimeFormatError>
+        where
+            ZP: ResourceProvider<provider::time_zones::TimeZoneFormatsV1Marker>
+            + ResourceProvider<provider::time_zones::ExemplarCitiesV1Marker>
+            + ResourceProvider<provider::time_zones::MetaZoneGenericNamesLongV1Marker>
+            + ResourceProvider<provider::time_zones::MetaZoneGenericNamesShortV1Marker>
+            + ResourceProvider<provider::time_zones::MetaZoneSpecificNamesLongV1Marker>
+            + ResourceProvider<provider::time_zones::MetaZoneSpecificNamesShortV1Marker>
+            + ?Sized,
+    {
+        load_resource(
+            &self.locale,
+            &mut self.data_payloads.mz_generic_long,
+            zone_provider,
+        )?;
+        if self.data_payloads.exemplar_cities == None {
+            load_resource(
+                &self.locale,
+                &mut self.data_payloads.exemplar_cities,
+                zone_provider,
+            )?;
+        }
+        self.units.push(TimeZoneFormatUnit::GenericNonLocationLongFormat(GenericNonLocationLongFormat{}));
+        Ok(self)
+    }
+
+    pub fn load_generic_non_location_short<ZP>(
+        &mut self,
+        zone_provider: &ZP,
+    ) -> Result<&mut TimeZoneFormat, DateTimeFormatError>
+        where
+            ZP: ResourceProvider<provider::time_zones::TimeZoneFormatsV1Marker>
+            + ResourceProvider<provider::time_zones::ExemplarCitiesV1Marker>
+            + ResourceProvider<provider::time_zones::MetaZoneGenericNamesLongV1Marker>
+            + ResourceProvider<provider::time_zones::MetaZoneGenericNamesShortV1Marker>
+            + ResourceProvider<provider::time_zones::MetaZoneSpecificNamesLongV1Marker>
+            + ResourceProvider<provider::time_zones::MetaZoneSpecificNamesShortV1Marker>
+            + ?Sized,
```

Comment:
The load functions should only have generic bounds for the exact data that they load.

In this case, it's only `MetaZoneGenericNamesShortV1Marker` and `ExemplarCitiesV1Marker`.

Reply:
sg! fixed it!